drivers/flash/Kconfig.nordic_qspi_nor

```
@@ -1,12 +1,15 @@
 # Copyright (c) 2019, Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
+DT_COMPAT_QSPI_NOR := nordic,qspi-nor
+
 menuconfig NORDIC_QSPI_NOR
 	bool "QSPI NOR Flash"
 	select FLASH_HAS_DRIVER_ENABLED
 	select NRFX_QSPI
 	select FLASH_JESD216
 	depends on HAS_HW_NRF_QSPI
+	default $(dt_compat_enabled,$(DT_COMPAT_QSPI_NOR))
```

Since `HAS_HW_NRF_QSPI` is defined as follows: https://github.com/zephyrproject-rtos/zephyr/blob/9680483c3d6aa91894326b0cad37004d56422638/soc/arm/nordic_nrf/Kconfig.peripherals#L129-L130
I'd suggest to use just:
```suggestion
	depends on HAS_HW_NRF_QSPI
	default y
```